sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/higherOrderFunctions.scala

```diff
@@ -72,16 +72,18 @@ object ResolveLambdaVariables extends Rule[LogicalPlan] {
     case LambdaFunction(function, names, _) =>
       if (names.size != argInfo.size) {
         e.failAnalysis(
-          errorClass = "_LEGACY_ERROR_TEMP_2300",
+          errorClass = "INVALID_LAMBDA_FUNCTION_CALL.NUM_ARGS_MISMATCH",
           messageParameters = Map(
-            "namesSize" -> names.size.toString,
-            "argInfoSize" -> argInfo.size.toString))
+            "expectedNumArgs" -> names.size.toString,
+            "actualNumArgs" -> argInfo.size.toString))
       }
 
       if (names.map(a => canonicalizer(a.name)).distinct.size < names.size) {
         e.failAnalysis(
-          errorClass = "_LEGACY_ERROR_TEMP_2301",
-          messageParameters = Map.empty)
+          errorClass = "INVALID_LAMBDA_FUNCTION_CALL.DUPLICATE_ARG_NAMES",
+          messageParameters = Map(
+            "args" -> names.map(a => canonicalizer(a.name)).mkString(", "),
+            "caseSensitiveConfig" -> "spark.sql.caseSensitive"))
```

Comment:
Could you wrap the SQL config by `toSQLConf` like:
```suggestion
            "caseSensitiveConfig" -> toSQLConf(SQLConf.CASE_SENSITIVE.key)))
```

Reply:
Got it.